packages/gtmcore/gtmcore/dataset/manifest/file.py

```diff
@@ -0,0 +1,288 @@
+from typing import List, Dict, Optional, NamedTuple
+import pickle
+import os
+from enum import Enum
+from collections import OrderedDict
+import json
+import redis
+import glob
+import copy
+
+from gtmcore.dataset.dataset import Dataset
+from gtmcore.logging import LMLogger
+
+logger = LMLogger.get_logger()
+
+
+class PersistTaskType(Enum):
+    """Enumeration of persist tasks"""
+    DELETE = 0
+    ADD = 1
+    UPDATE = 2
+
+
+# NamedTuple to capture tasks to be done to persist manifest changes to disk
+PersistTask = NamedTuple('PersistTask', [('relative_path', str), ('task', PersistTaskType), ('manifest_file', str)])
+
+
+class ManifestJSONEncoder(json.JSONEncoder):
+    """A custom json encoder to output json files that will be moderately compatible with git"""
+    def encode(self, o):
+        if isinstance(o, OrderedDict):
+            output = [f'"{x}":{json.dumps(o[x], separators=(",", ":"))}' for x in o]
+            return "{\n" + ",\n".join(output) + "\n}"
+
+        else:
+            return json.dumps(o)
+
+
+class ManifestFileCache(object):
+    """Class to provide a caching layer on top of a collection of Dataset manifest files
+
+    Note: The checkout context of the underlying dataset CANNOT change while this class is instantiated. If it does,
+    you need to reload the Dataset instance and reload the Manifest instance.
+
+    """
+    def __init__(self, dataset: Dataset, logged_in_username: Optional[str] = None) -> None:
+        self.dataset = dataset
+        self.logged_in_username = logged_in_username
+
+        self.ignore_file = os.path.join(dataset.root_dir, ".gigantumignore")
+
+        self._redis_client = None
+        self._manifest: OrderedDict = OrderedDict()
+        self._current_checkout_id = self.dataset.checkout_id
+        self._persist_queue: List[PersistTask] = list()
+
+        # TODO: Support ignoring files
```

Comment:
Here and elsewhere, claim your TODOs?

```suggestion
        # DMK TODO: Support ignoring files
```

Reply:
while I may do the work, anyone could pick this up. I suppose should be a ticket vs. a TODO, but wanted to mark where you'd hook things up. Will create a ticket at least.